HW #5 Application architecture

cholombytko opened the pull request:
This pull request contains all changes which will occure during hw5 solving

cholombytko committed:
feat: implemented architecture diagram

This commit contains implementation of architecture file in docx format

eluyema commented:
Все чітко і зрозуміло, дорікань в мене немає

eluyema reviewed (approved):
Все чітко і зрозуміло, дорікань в мене немає

cholombytko committed:
Merge branch 'hw4/gof' of github.com:GenesisEducationKyiv/software-engineering-school-4-0-cholombytko into hw5/appl-architecture

cholombytko committed:
feat: new application architecture diagram

This commit contains new version of application architecture diagram

cholombytko committed:
feat: architecture test

This commit contains architecture test implementation

eluyema commented on appl-architecture-v2.docx:
А чому в тебе два файла docx?

cholombytko replied:
Forgot to remove previous version, ty!

eluyema commented on test/architecture.spec.ts:
```diff
@@ -0,0 +1,17 @@
+describe('architecture', () => {
+  it('application layer should not depend on ui', () => {
+    const uiPattern = '/.*.controller.ts$/i';
+    const servicePattern = '/.*.service.ts$/i';
```
Цікаво, однак не хочеш ще для репозиторію це зробити ?

cholombytko replied:
Great idea!

cholombytko committed:
refactor: delete old version of architecture

This commit contains project refactoring by deleting old version of
application architecture diagram

Larisa-Antsifrova commented:
Thank you for your work! 

I would also suggest adding some dependency flow arrows to the diagram to see how different parts of the system depend on each other.

Also, I am not quite sure if the current application follows any of the architecture patterns :) If yes, let me know which.
